### ISSUE TITLE
write instructions for building and installing documentation

### ISSUE THREAD

**Issue by marcomaggi** [cisco/ChezScheme#209]
The file `BUILDING` has no instructions about how to build and install documentation.  With a checkout up to commit `b79099636e5ef82c8f80e56f5aede3a90d0c1f39` if I do `make docs` I get this:
```
$make docs
(cd a6le && make build)
(cd c ; make)
make[2]: Nothing to be done for 'doit'.
(cd s ; make bootstrap)
make[2]: 'bootstrap' is up to date.
(cd csug && make m=a6le)
../a6le/bin/a6le/scheme -b ../a6le/boot/a6le/petite.boot -b ../a6le/boot/a6le/scheme.boot --libdirs "../stex/src" --program ../stex/src/scheme-prep.ss tspl4-prep csug
Exception: library (dsm) not found
../stex/Mf-stex:41: recipe for target 'csug.tex' failed
make[1]: *** [csug.tex] Error 255
Makefile:31: recipe for target 'docs' failed
make: *** [docs] Error 2
```

I have not looked at the code yet, would it be feasible to write a back end for the documentation system to produce a different output format?  Was the system written for this?

**Comment by akeep** [comment]
Yes, we should probably add some instructions, but in general ``make docs`` should just work.

The library ``dsm.ss`` should be in the ``stex`` submodule that lands in the ``stex`` directory, so the ``--libdirs "../stex/src"`` should be able to find it.  Is it possible you have set ``CHEZSCHEMELIBEXTS`` to something that does not include ``ss`` before trying the build?

**Comment by marcomaggi** [comment]
Yes, this bites me again and again.  But now I get this with the latest commit:

```
...
use canned/cisco-logo-large.png> [1{/usr/local/texlive/2016/texmf-var/fonts/ma
p/pdftex/updmap/pdftex.map} <./canned/cisco-logo-large.png>]) (./copyright.tex
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/omscmr.fd)
! Undefined control sequence.
l.28 Revised \INSERTREVISIONMONTHSPACEYEAR
                                           for Chez Scheme Version 9.4.1.
?
```

using this build script:

```
#!/bin/bash
#
# Informal build script for Chez Scheme

export CHEZSCHEMELIBEXTS=.chezscheme.sls:.chezscheme.ss:.sls:.ss
#export CHEZSCHEMELIBDIRS=:
unset CHEZSCHEMELIBDIRS

PREFIX=/opt/chez-scheme/9.4.1${CHEZ_INSTALL_SUFFIX}

make clean
make distclean
git clean -xfd
./configure \
    --installprefix="$PREFIX"			\
    --installowner=root				\
    --installgroup=root				\
    --installschemename=chez			\
    --installpetitename=petite-chez		\
    CFLAGS='-O3 -Wno-maybe-uninitialized'	\
    CPPFLAGS='-I/usr/local/include'		\
    LDFLAGS='-L/usr/local/lib64 -liconv'
make
make docs

### end of file
```

**Comment by marcomaggi** [comment]
Still fails with commit `c9c45641cc50b073329cc6d219140cf93fbd5c4c`.  Do I need to install a git hook or something like that?

**Comment by akeep** [comment]
Sorry I did not get this in before the release.  The ``\INSERTREVISIONMONTHSPACEYEAR`` tag in there was something that needed to be replaced by hand when the documents are generated.  I have replaced this with a little code to put the current month and year in the corresponding place in the document and it should build without needing the hand editing as of commit 1febe9b.

**Comment by marcomaggi** [comment]
Yes!  Build succeeds with commit `1febe9b1a1e1cde4b7be8bc562153ee4b42afd8c`.  Now, ahem, is there no `make install-doc` rule?

**Comment by marcomaggi** [comment]
I noticed that, after running `make docs`, if I do a `git status` I see a number of files not tracked by git, but not ignored through `.gitignore`.  A purist would say that this should not happen...

**Comment by akeep** [comment]
Thanks for the reminder @marcomaggi, I've added the finished documents and intermediate files to the ``.gitignore``

**Comment by akeep** [comment]
As far as an ``install-docs`` rule in the makefile, the current ``install`` writes these documents to ``/tmp`` in preparation for putting them online.  So, before we add an ``install-docs`` we probably need to decide where these should go.

For unix-like operating systems I think putting these in ``/usr/share/csv<version_number>`` or ``/usr/local/share/csv<version_number>`` is probably the natural default for these (we could put this under the ``installprefix`` directory and add an ``installdocsuffix`` for controlling this under the ``installprefix`` as we do with  the manual pages.

I'm not sure about Windows INSTALL, though for cygwin we could certainly follow the same convention.

**Comment by marcomaggi** [comment]
On Unix-like systems I suggest `$prefix/share/doc/csv<version_number>`: that's where stuff goes on my Slackware installation, and it is also the one selected by the GNU Autotools (minus the version number).  I cannot help for Windows.

**Comment by marcomaggi** [comment]
I correct myself, the doc directory pathname should be `$prefix/share/doc/csv-<version_number>` or the more explicit `$prefix/share/doc/ChezScheme-<version_number>`, with a dash between the package name and the version number.  This is so for Slackware.
